src/NLog/Config/StackTraceUsage.cs

```diff
@@ -33,36 +33,56 @@
 
 namespace NLog.Config
 {
+    using System;
+
     /// <summary>
     /// Value indicating how stack trace should be captured when processing the log event.
     /// </summary>
+    [Flags]
     public enum StackTraceUsage
     {
         /// <summary>
         /// Stack trace should not be captured.
         /// </summary>
-        None = 0, 
+        None = 0,
+
+        /// <summary>
+        /// Stack trace should be captured
+        /// </summary>
+        WithStackTrace = 1,
+
+#if !SILVERLIGHT
+        /// <summary>
+        /// Source-level information should be capture (source-filename + linenumber)
+        /// </summary>
+        WithSourceCode = 2,
```

Comment:
The big reason why this isn't merged yet are the changed write method and the names of this enum.

I'm doubting if sourcecode is clear. But for now haven't better names

Reply:
Maybe rename this to WithFileNameAndLineNumber?

Reply:
Fine with me